### PR TITLE
Update API documentation

### PR DESCRIPTION
**Pull request by hubertmis**

### CODE REVIEW

File: src/nrf_802154_config.h [hunk 1]
```diff
@@ -97,8 +97,7 @@ extern "C" {
  *
  * If the driver should internally handle the RADIO IRQ.
  * In case the driver is used in an OS the RADIO IRQ may be handled by the OS and passed to
- * the driver @sa nrf_802154_irq_handler(). In this case internal handling should be
- * disabled.
+ * the driver @sa nrf_802154_irq_handler(). In this case internal handling should be disabled.
  */
 
 #ifndef NRF_802154_INTERNAL_IRQ_HANDLING
```
Comment (LuDuda) [inline src/nrf_802154_config.h]: Let's leave it.. But i guess, after some thoughts.. i would prefer the opposite handling e.g.:
NRF_802154_ENABLE_EXTERNAL_RADIO_IRQ_HANDLER

File: src/nrf_802154_config.h [hunk 2]
```diff
@@ -97,8 +97,7 @@ extern "C" {
  *
  * If the driver should internally handle the RADIO IRQ.
  * In case the driver is used in an OS the RADIO IRQ may be handled by the OS and passed to
- * the driver @sa nrf_802154_irq_handler(). In this case internal handling should be
- * disabled.
+ * the driver @sa nrf_802154_irq_handler(). In this case internal handling should be disabled.
```
Comment (LuDuda) [inline src/nrf_802154_config.h]: Shouldn't we name it nrf_802154_radio_irq_handler() ?

File: src/nrf_802154_config.h [hunk 3]
```diff
@@ -169,10 +180,25 @@ extern "C" {
...
 #endif
 
+/**
+ * @def NRF_802154_NOTIFICATION_SWI_PRIORITY
+ *
+ * Priority of software interrupt used for requests and notifications.
+ *
+ */
+#ifndef NRF_802154_EGU_PRIORITY
+#define NRF_802154_EGU_PRIORITY 5
```
Comment (KaSroka) [inline src/nrf_802154_config.h]: Why section is called SWI_PRIORITY but define is called EGU_PRIORITY?

File: src/nrf_802154_config.h [hunk 4]
```diff
@@ -169,10 +180,25 @@ extern "C" {
...
+#endif
+
 /**
  * @def NRF_802154_USE_RAW_API
  *
- * If RAW or copying API should be available for the MAC layer.
+ * If RAW API should be available for the MAC layer.
```
Comment (LuDuda) [inline src/nrf_802154_config.h]: I guess we should combine sentence 1 with 2. The first sentence itself does not have any sence.

File: src/nrf_802154_config.h [hunk 5]
```diff
@@ -208,33 +234,12 @@ extern "C" {
...
-#ifndef NRF_802154_NOTIFICATION_SWI_PRIORITY
-#define NRF_802154_NOTIFICATION_SWI_PRIORITY 5
-#endif
-
 /**
  * @def NRF_802154_EXCLUDE_BCC_MATCHING
  *
- * Setting this flag will disable NRF_RADIO_EVENT_BCMATCH handling, and therefore address filtering during frame reception.
- * With this flag set to 1, address filtering is done after receiving a frame, during NRF_RADIO_EVENT_END handling.
+ * Setting this flag will disable NRF_RADIO_EVENT_BCMATCH handling, and therefore address filtering
```
Comment (LuDuda) [inline src/nrf_802154_config.h]: will disable -> disables?

File: src/nrf_802154_config.h [hunk 6]
```diff
@@ -449,7 +443,7 @@ extern "C" {
  *       These features depend on notification of transmission start.
  */
 #ifndef NRF_802154_TX_STARTED_NOTIFY_ENABLED
-#if NRF_802154_ACK_TIMEOUT_ENABLED ||                                                     \
+#if NRF_802154_ACK_TIMEOUT_ENABLED ||                                                              \
```
Comment (LuDuda) [inline src/nrf_802154_config.h]: Can we just merge those two line into one?

Comment (LuDuda) [reply src/nrf_802154_config.h]: #if NRF_802154_ACK_TIMEOUT_ENABLED || NRF_802154_CSMA_CA_ENABLED

File: src/nrf_802154_swi.c [hunk 7]
```diff
@@ -394,12 +394,9 @@ void nrf_802154_swi_init(void)
...
     m_ntf_w_ptr = 0;
 
-    nrf_egu_int_enable(SWI_EGU,
-                       NTF_INT |
-                       TIMESLOT_EXIT_INT |
-                       REQ_INT);
+    nrf_egu_int_enable(SWI_EGU, NTF_INT | TIMESLOT_EXIT_INT |  REQ_INT);
 
-    NVIC_SetPriority(SWI_IRQn, NRF_802154_NOTIFICATION_SWI_PRIORITY);
+    NVIC_SetPriority(SWI_IRQn, NRF_802154_EGU_PRIORITY);
```
Comment (KaSroka) [inline src/nrf_802154_swi.c]: Why we keep SWI IRQ priority in EGU_PRIORITY?

File: src/nrf_802154_config.h (hunk 3 above)
Comment (hubertmis) [reply src/nrf_802154_config.h]: SWI and EGU are tightly coupled in nRF52. But you are probably right that priority is for SWI. EGU does not handle interrupts in software.

Comment (hubertmis) [reply src/nrf_802154_config.h]: OK, I see what you mean. Corrected.

**Pull request merged**